### PR TITLE
SQL Migrations for warehouse

### PR DESCRIPTION
**Pull request by fxenik**
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-server/365)

### CODE REVIEW

File: warehouse/warehouse.go [hunk 1]
```diff
@@ -87,6 +83,15 @@ const (
...
 
+// warehouse table names
+const (
+	warehouseStagingFilesTable = "wh_staging_files"
+	warehouseLoadFilesTable    = "wh_load_files"
+	warehouseUploadsTable      = "wh_uploads"
+	warehouseTableUploadsTable = "wh_table_uploads"
+	warehouseSchemasTable      = "wh_schemas"
+)
+
```
Comment (psrikanth88) [inline warehouse/warehouse.go:94]: let's remove this in config.toml also and use from constants in other files like warehouseutils also?

File: sql/migrations/warehouse/000001_create_tables.up.sql [hunk 2]
```diff
@@ -0,0 +1,139 @@
+
+--
+-- wh_staging_files
+--
+
+DO $$ BEGIN
+    DROP TYPE wh_staging_state_type;
```
Comment (snarkychef) [inline sql/migrations/warehouse/000001_create_tables.up.sql:7]: This should be after `ALTER TABLE wh_staging_files ALTER COLUMN status TYPE VARCHAR(255);` ?

File: sql/migrations/warehouse/000001_create_tables.up.sql [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+ALTER TABLE wh_load_files ADD COLUMN IF NOT EXISTS total_events BIGINT;
+
+CREATE INDEX IF NOT EXISTS wh_load_files_source_destination_id_table_name_index ON wh_load_files (source_id, destination_id, table_name);
+
+--
+-- wh_uploads
+--
+
+DO $$ BEGIN
+    DROP TYPE wh_upload_state_type;
```
Comment (snarkychef) [inline sql/migrations/warehouse/000001_create_tables.up.sql:60]: This should be afrer `ALTER TABLE wh_uploads ALTER COLUMN status TYPE VARCHAR(255);` ?

File: sql/migrations/warehouse/000001_create_tables.up.sql [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+CREATE INDEX IF NOT EXISTS wh_uploads_source_destination_id_index ON wh_uploads (source_id, destination_id);
+
+ALTER TABLE wh_uploads ALTER COLUMN status TYPE VARCHAR(255);
+
+--
+-- wh_table_uploads
+--
+
+DO $$ BEGIN
+    DROP TYPE wh_table_upload_state_type;
```
Comment (snarkychef) [inline sql/migrations/warehouse/000001_create_tables.up.sql:101]: This should be after `ALTER TABLE wh_table_uploads ALTER COLUMN status TYPE VARCHAR(255);`?

File: sql/migrations/warehouse/000001_create_tables.up.sql [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    id BIGSERIAL PRIMARY KEY,
+    wh_upload_id BIGSERIAL,
+    source_id VARCHAR(64) NOT NULL,
+    namespace VARCHAR(64) NOT NULL,
+    destination_id VARCHAR(64) NOT NULL,
+    destination_type VARCHAR(64) NOT NULL,
+    schema JSONB NOT NULL,
+    error TEXT,
+    created_at TIMESTAMP NOT NULL);
+
```
Comment (snarkychef) [inline sql/migrations/warehouse/000001_create_tables.up.sql:138]: `DROP INDEX IF EXISTS wh_schemas_source_destination_id_index;` missing?

File: warehouse/warehouse.go (hunk 1 above)
Comment (snarkychef) [reply warehouse/warehouse.go:94]: Yeah. we should do this.

File: sql/migrations/warehouse/000001_create_tables.up.sql [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    id BIGSERIAL PRIMARY KEY,
+    source_id VARCHAR(64) NOT NULL,
+    namespace VARCHAR(64) NOT NULL,
+    destination_id VARCHAR(64) NOT NULL,
+    destination_type VARCHAR(64) NOT NULL,
+    start_staging_file_id BIGINT,
+    end_staging_file_id BIGINT,
+    start_load_file_id BIGINT,
+    end_load_file_id BIGINT,
+    status VARCHAR(255) NOT NULL,
```
Comment (snarkychef) [inline sql/migrations/warehouse/000001_create_tables.up.sql:74]: Let's use `VARCHAR(64)` for all status types?

**Pull request merged**